hera_cal/lst_stack/calibration.py

```diff
@@ -100,16 +100,26 @@ def _expand_degeneracies_to_ant_gains(
                 )
                 fmats[split_pol1].append(fmat)
 
+    # Dictionary for storing gains
     gains = {}
-    for polidx, pol in enumerate(stack.pols):
-        split_pol1, split_pol2 = utils.split_pol(pol)
-        if split_pol1 != split_pol2:
-            continue
+
+    # Get flags common to all baselines
+    flags = np.all(stack.flags, axis=1)
+
+    # Map antenna-polarizations to visibility indices
+    antpol_to_idx = {
+        utils.split_pol(pol)[0]: polidx
+        for polidx, pol in enumerate(stack.pols) 
+        if utils.split_pol(pol)[0] == utils.split_pol(pol)[1]
+    }
+    unique_pols = list(antpol_to_idx.keys())
+
+    for polidx, pol in enumerate(unique_pols):
```

Comment:
Is there a reason you're not looping over antpol_to_idx.items()

Reply:
No, you're right, that makes more sense. Fixing this now